signer/tests/integration/transaction_signer.rs

```diff
@@ -0,0 +1,72 @@
+use signer::storage;
+use signer::testing;
+
+use futures::StreamExt;
+use rand::RngCore;
+use sqlx::Executor;
+
+async fn test_environment(
+    pool: sqlx::PgPool,
+) -> testing::transaction_signer::TestEnvironment<impl FnMut() -> storage::postgres::PgStore> {
+    let num_signers = 3;
+    let context_window = 3;
+    let test_databases: Vec<_> = futures::stream::iter(0..num_signers)
+        .then(|_| async { new_database(&pool).await })
+        .collect()
+        .await;
+
+    let mut idx = 0;
+
+    testing::transaction_signer::TestEnvironment {
+        storage_constructor: move || {
+            idx = (idx + 1) % test_databases.len();
+            storage::postgres::PgStore::from(test_databases.get(idx).unwrap().clone())
+        },
+        context_window,
+        num_signers,
+    }
+}
+
+static MIGRATOR: sqlx::migrate::Migrator = sqlx::migrate!();
+
+async fn new_database(pool: &sqlx::PgPool) -> sqlx::PgPool {
+    let mut rng = rand::rngs::OsRng;
+    let db_name = format!("test_db_{}", rng.next_u64());
+
+    let create_db = format!("CREATE DATABASE \"{}\";", db_name);
```

Comment:
nit: `format!("CREATE DATABASE \"{db_name}\";");`

Reply:
Thanks for spotting. I tend to forget that we can do this nowadays :sweat_smile: It was one of the features I missed coming from Python in the early days.

Reply:
Heck, I still miss it all the time too.